src/CoreWCF.Primitives/src/CoreWCF/Security/Authorization/AuthorizationServiceBehavior.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Reflection;
+using CoreWCF.Channels;
+using CoreWCF.Description;
+using Microsoft.AspNetCore.Authorization;
+
+namespace CoreWCF.Security.Authorization
+{
+    public class AuthorizationServiceBehavior : IServiceBehavior
+    {
+        private readonly IAuthorizationPolicyProvider _authorizationPolicyProvider;
+        private readonly IAuthorizationService _authorizationService;
+
+        public AuthorizationServiceBehavior(IAuthorizationPolicyProvider authorizationPolicyProvider,
+            IAuthorizationService authorizationService)
+        {
+            _authorizationPolicyProvider = authorizationPolicyProvider;
+            _authorizationService = authorizationService;
+        }
+
+        public void Validate(ServiceDescription serviceDescription, ServiceHostBase serviceHostBase)
+        {
+
+        }
+
+        public void AddBindingParameters(ServiceDescription serviceDescription, ServiceHostBase serviceHostBase,
+            Collection<ServiceEndpoint> endpoints,
+            BindingParameterCollection bindingParameters)
+        {
+
+        }
+
+        public void ApplyDispatchBehavior(ServiceDescription serviceDescription, ServiceHostBase serviceHostBase)
+        {
+            foreach (var endpoint in serviceDescription.Endpoints)
+            {
+                foreach (var operation in endpoint.Contract.Operations)
+                {
+                    MethodInfo methodInfo = GetServiceImplMethodInfo(serviceDescription, operation);
+                    bool hasAllowAnonymous = methodInfo.GetCustomAttribute(typeof(AllowAnonymousAttribute)) is not null;
+                    var authorizeData = methodInfo
+                        .GetCustomAttributes(typeof(AuthorizeAttribute), false).OfType<IAuthorizeData>()
+                        .ToArray();
+
+                    operation.OperationBehaviors.Add(
+                        new AuthorizationOperationBehavior(authorizeData, _authorizationPolicyProvider,
+                            _authorizationService, hasAllowAnonymous));
+
+                }
+            }
+        }
+
+        private static MethodInfo GetServiceImplMethodInfo(ServiceDescription serviceDescription, OperationDescription operationDescription)
+        {
+            Type serviceType = serviceDescription.ServiceType;
+            MethodInfo[] methodInfos = serviceType.GetMethods();
+            return methodInfos.First(x => x.Name == operationDescription.Name);
+        }
```

Comment:
This isn't going to work as the operationDescription.Name is often not the same as the method name. There's some mapping we do. For example BeginFoo -> Foo, and BarAsync -> Bar. On top of that, the name might come from OperationContractAttribute.Name where you can explicitly provide the name. There's an internal property on OperationDescription which will provide you the MethodInfo which will get called for this operation. It's called OperationMethod.

Reply:
`OperationMethod` property is now used.